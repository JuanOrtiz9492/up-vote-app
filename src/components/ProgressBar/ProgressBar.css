progress[value] {
    /* Reset the default appearance */
    -webkit-appearance: none;
     appearance: none;
  
    width: 250px;
    height: 30px;
  }

  progress[value]::-webkit-progress-bar {
    background-color: #b37a15;
    border-radius: 2px;
    box-shadow: 0 2px 5px rgba(0, 0, 0, 0.25) inset;
  }
  progress[value]::-webkit-progress-value {
    
    background-color: #14847f;

  }
  progress[value]::-webkit-progress-value::before {
    content: '80%';
    position: absolute;
    right: 0;
    top: -125%;
  }
  progress[value]::-webkit-progress-value::after {
    content: '';
    width: 6px;
    height: 6px;
    position: absolute;
    border-radius: 100%;
    right: 7px;
    top: 7px;
    background-color: white;
  }