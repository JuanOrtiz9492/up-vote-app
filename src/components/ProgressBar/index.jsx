import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types'

import { constants } from '../../config/constants';
import { cssConstants } from '../../config/cssConstants';
import {ThumbsUpIcon, ThumbsDownIcon} from '../Icon/IconList';


const { fire, jungle } = cssConstants.colors;

const StyledProgressBar = styled.progress`
    position: absolute;
    top:0px;
    left: 0px;

    &[value] {
    -webkit-appearance: none;
    appearance: none;
    width: ${cssConstants.progressBar.width};
    height: 3em;

    &::-webkit-progress-bar {
        background-color: ${fire.hex};
        border-radius: 2px;
        box-shadow: 0 2px 5px rgba(0, 0, 0, 0.25) inset;
      }

    &::-webkit-progress-value {
        background-color: ${jungle.hex};    
    }
  }
`

const StyledContainer = styled.div`
    position: relative;
    width: ${cssConstants.progressBar.width};
`

const StyledPercentage = styled.div`
    position: absolute;
    top:0px;
    position: absolute;
    z-index: 10;
    margin 0 0.5em;
    &.left {
        left: 0px;
    }
    &.right {
        right: 0px;
    }
`
const StyledLabel = styled.span`
    font-size: 2em;
`

const ProgressBar = ({value, maxValue, ...props}) => (
    <StyledContainer className="progress-bar">
        <StyledPercentage className="left">
            <ThumbsUpIcon width={1.5}/>
            <StyledLabel>{`${value}%`}</StyledLabel>
        </StyledPercentage>
        <StyledPercentage className="right">
            <StyledLabel>{`${maxValue-value}%`}</StyledLabel>
            <ThumbsDownIcon width={1.5}/>
        </StyledPercentage>
        <StyledProgressBar value={value} max={maxValue}/>
    </StyledContainer>
    )

ProgressBar.propTypes = {
    value: PropTypes.number,
    maxValue: PropTypes.number,
}

ProgressBar.defaultProps = {
    value: 50,
    maxValue: 100,
}

export default ProgressBar;