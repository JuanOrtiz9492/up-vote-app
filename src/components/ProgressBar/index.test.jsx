import React from 'react';
import { render } from '@testing-library/react';
import ProgressBar from '.';

describe('ProgressBar Component', () => {

    it('should Match the Snapshot', () => {
        const { container } = render(<ProgressBar />);
        expect(container).toMatchSnapshot();
      })
})

