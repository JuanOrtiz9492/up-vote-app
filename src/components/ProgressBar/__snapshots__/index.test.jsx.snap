// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar Component should Match the Snapshot 1`] = `
<div>
  <div
    class="sc-AykKJ fNCYSY progress-bar"
  >
    <div
      class="sc-AykKK gPvMTr left"
    >
      <svg
        class="sc-AykKC iUTaVw"
        color="#ffffff"
        width="1.5"
      >
        thumbs-up.svg
      </svg>
      <span
        class="sc-fzXfLO iMHASE"
      >
        50%
      </span>
    </div>
    <div
      class="sc-AykKK gPvMTr right"
    >
      <span
        class="sc-fzXfLO iMHASE"
      >
        50%
      </span>
      <svg
        class="sc-AykKD gSoynF"
        color="#ffffff"
        width="1.5"
      >
        thumbs-down.svg
      </svg>
    </div>
    <progress
      class="sc-AykKI cRdaSl"
      max="100"
      value="50"
    />
  </div>
</div>
`;
